### PR TITLE
IIP-13 / Feature  Add inline actions for impex files

### PR DESCRIPTION
**Pull request by vladyanytskyi**
Added ability to copy text from Impex and Flexible Search files to Hybris Console.

Signed-off-by: Vladyslav Yanytskyi vladyslav_yanytskyi@epam.com

### CODE REVIEW

File: src/com/intellij/idea/plugin/hybris/flexibleSearch/file/actions/CopyFlexibleSearchFileAction.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.TreePath;
+
+public class CopyFlexibleSearchFileAction extends AbstractCopyFileToHybrisConsoleAction {
+
+    private static final String FLEXIBLE_SEARCH_CONSOLE_TITLE = "Hybris FS Console";
+
+    @Override
+    public void update(@NotNull final AnActionEvent e) {
+        final DataContext dataContext = e.getDataContext();
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/flexibleSearch/file/actions/CopyFlexibleSearchFileAction.java]: Why do we need to have `DataContext dataContext` if `Project project = e.getProject();`?

File: src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        final String fileExtension,
+        final boolean oneFile
+    ) {
+        boolean isImpex = false;
+        final List<String> filesName = getFilesName(dataContext);
+        if (oneFile) {
+            return filesName.size() == 1 && filesName.get(0).contains(fileExtension);
+        } else {
+            for (String name : filesName) {
+                if (name.contains(fileExtension)) {
```
Comment (MykytaKostiuk) [inline src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java:55]: We need to have an access to VirtualFile and then check its extension.
Smth like
```
StructureTreeModel.Node lastPathComponent = (StructureTreeModel.Node) file.getLastPathComponent();
ProjectViewNode element = (ProjectViewNode) lastPathComponent.getElement();
element.getVirtualFile().getExtension();
```

File: src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                    isImpex = true;
+                } else {
+                    return false;
+                }
+            }
+        }
+        return isImpex;
+    }
+
+    private List<String> getFilesName(@NotNull final DataContext dataContext) {
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java]: We need to have an access to VirtualFile and then check its extension.
Smth like
```
StructureTreeModel.Node lastPathComponent = (StructureTreeModel.Node) file.getLastPathComponent();
ProjectViewNode element = (ProjectViewNode) lastPathComponent.getElement();
element.getVirtualFile().getExtension();
```

File: src/com/intellij/idea/plugin/hybris/flexibleSearch/file/actions/CopyFlexibleSearchFileAction.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Override
+    public void actionPerformed(@NotNull final AnActionEvent e) {
+        final Project project = e.getProject();
+        final TreePath[] files = (TreePath[]) getFiles(e.getDataContext());
+        for (final TreePath treePath : files) {
+            final DefaultMutableTreeNode lastPathNode = (DefaultMutableTreeNode) treePath.getLastPathComponent();
+            final PsiFileNode file = (PsiFileNode) lastPathNode.getUserObject();
+            final HybrisConsolePanel hybrisConsolePanel = getHybrisConsolePanel(project);
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/flexibleSearch/file/actions/CopyFlexibleSearchFileAction.java]: if you want to check whether the project is not null - then it should be before this line

File: src/com/intellij/idea/plugin/hybris/flexibleSearch/file/actions/CopyFlexibleSearchFileAction.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            final HybrisConsole hybrisConsole = hybrisConsolePanel.findConsole(FLEXIBLE_SEARCH_CONSOLE_TITLE);
+            if (hybrisConsole != null && project != null) {
+                if (!getTextFromHybrisConsole(project, hybrisConsole).isEmpty()) {
+                    final CopyFileToHybrisConsoleDialog copyFileToHybrisConsoleDialog = new CopyFileToHybrisConsoleDialog(project);
+                    copyFileToHybrisConsoleDialog.setTitle("Flexible Search Console");
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/flexibleSearch/file/actions/CopyFlexibleSearchFileAction.java]: string label is a good candidate to be moved to the property file

File: src/com/intellij/idea/plugin/hybris/impex/file/actions/CopyImpexFileAction.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public void update(@NotNull final AnActionEvent e) {
+        final DataContext dataContext = e.getDataContext();
+        e.getPresentation()
+         .setEnabledAndVisible(ActionUtils.isHybrisContext(dataContext) && isRequiredFileExtension(
+             dataContext, ".impex", false));
+    }
+
+    @Override
+    public void actionPerformed(@NotNull final AnActionEvent e) {
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/impex/file/actions/CopyImpexFileAction.java]: Seems that common code could be also moved to AbstractCopyFileToHybrisConsoleAction

File: src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+        final Object[] files = getFiles(dataContext);
+        final List<String> names = new ArrayList<>();
+        if (files != null) {
+            for (final Object file : files) {
+                final TreePath treePath = (TreePath) file;
+                names.add(treePath.getLastPathComponent().toString());
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java]: Don't trust to *.toString()

File: src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+            }
+        }
+        return names;
+    }
+
+    public Object[] getFiles(@NotNull final DataContext dataContext) {
```
Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java]: Try not to return the `Object` or `Object[]`
Cast to the specific type which will be returned.  like `TreePath`

Comment (ghost) [inline src/com/intellij/idea/plugin/hybris/actions/AbstractCopyFileToHybrisConsoleAction.java]: here we can return array or null (`.getSelectionPaths()`), is it possible to return just an array, populated or empty?

**Pull request merged**